client/verta/verta/registry/entities/_modelversion.py

```diff
@@ -1514,3 +1519,59 @@ def set_hide_output_label(self, hide):
     def get_hide_output_label(self):
         self._refresh_cache()
         return self._msg.hide_output_label
+
+    def log_dataset_version(self, key, dataset_version):
+        """
+        Logs a Verta DatasetVersion to this Model Version with the given key.
+
+        Parameters
+        ----------
+        key : str
+            Name of the dataset version.
+        dataset_version : :class:`~verta.dataset.entities.DatasetVersion`
+            Dataset version.
+
+        """
+        if not isinstance(dataset_version, _dataset.DatasetVersion):
+            raise TypeError("`dataset_version` must be of type DatasetVersion")
+
+        artifact_msg = _CommonCommonService.Artifact(
+            key=key,
+            path_only=True,
+            artifact_type=_CommonCommonService.ArtifactTypeEnum.DATA,
+            linked_artifact_id=dataset_version.id,
+        )
+
+        msg = _RegistryService.LogDatasetsInModelVersion(
+            model_version_id=self.id,
+            datasets=[artifact_msg],
+        )
+
+        endpoint = "/api/v1/registry/model_versions/{}/logDatasets".format(
+            self.id,
+        )
+
+        response = self._conn.make_proto_request("POST", endpoint, body=msg)
+        self._conn.must_response(response)
+        self._clear_cache()
+
+    def get_dataset_version(self, key):
+        """
+        Gets the DatasetVersion with name `key` from this Model Version.
+
+        Parameters
+        ----------
+        key : str
+            Name of the dataset version.
+
+        Returns
+        -------
+        `DatasetVersion <dataset.html>`_
+            DatasetVersion associated with the given key.
+
+        """
+        for dataset in self._msg.datasets:
+            if dataset.key == key:
+                return dataset
```

Comment:
I'm not 100% sure this works exactly, but based on [`ExperimentRun.get_dataset()`](https://github.com/VertaAI/modeldb/blob/b1c01068ca1abae05c8126f7fd78e84660555ad9/client/verta/verta/tracking/entities/_experimentrun.py#L1222-L1227):

```suggestion
                return _dataset_version.DatasetVersion(
                    self._conn,
                    self._conf,
                    _dataset_version.DatasetVersion._get_proto_by_id(
                        self._conn, dataset.linked_artifact_id
                    ),
                )
```

and then _I think_ for the test we can check `dataset.id` instead of `dataset.linked_artifact_id`